src/components/views/settings/devices/useOwnDevices.ts

```diff
@@ -176,6 +171,15 @@ export const useOwnDevices = (): DevicesState => {
         refreshDevices();
     }, [refreshDevices]);
 
+    useEffect(() => {
+        const deviceIds = Object.keys(devices);
+        // empty devices means devices have not been fetched yet
+        // as there is always at least the current device
+        if (deviceIds.length) {
+            pruneClientInformation(Object.keys(devices), matrixClient);
```

Comment:
One `Object.keys` call can be saved here:

```suggestion
            pruneClientInformation(deviceIds, matrixClient);
```